vmd-server/src/server/api.rs

```diff
@@ -0,0 +1,85 @@
+// === Api ====================================================================
+//
+// This module contains API hooks. The API definitions are generated by the
+// openapi-generator. The generated request handlers call the functions in this
+// module in order to query the server state and to perform actions. In case
+// the API definitions change, this module needs to be updated accordingly.
+//
+// The API interface is provided as a trait `Api<C>` which is implemented for
+// the `ApiImpl<C>` struct. The trait is generic over a context type `C` which
+// is used to pass additional information to the API handlers.
+//
+// === External crates ========================================================
+
+use swagger::{
+    Has,
+    XSpanIdString,
+    ApiError,
+};
+use async_trait::async_trait;
+use std::marker::PhantomData;
+use log::info;
+
+// === Internal modules =======================================================
+
+use vmd_rust_server_api::{
+    Api,
+    GetVmInfoByIdResponse,
+    GetVmListResponse,
+    VmActionResponse,
+};
+
+// === Implementations ========================================================
+
+#[derive(Copy, Clone)]
+pub struct ApiImpl<C> {
+    marker: PhantomData<C>,
+}
+
+impl<C> ApiImpl<C> {
+    pub fn new() -> Self {
+        ApiImpl{marker: PhantomData}
+    }
+}
+
+#[async_trait]
+impl<C> Api<C> for ApiImpl<C> where C: Has<XSpanIdString> + Send + Sync
+{
+    /// Gets virtual machine info by ID
```

Comment:
Not big fan of obvious comments which replicate the function name when the function name already provides great semantics for the reader.

Reply:
These are remnants from the generated API. I will remove them.

Reply:
https://github.com/tiiuae/vmd/pull/35/commits/5269609b05a6fe3231572e560c0298900206caee